src/systems/implicit_system.C

```diff
@@ -225,6 +226,18 @@ SparseMatrix<Number> & ImplicitSystem::add_matrix (const std::string & mat_name)
 }
 
 
+void ImplicitSystem::remove_matrix (const std::string & mat_name)
+{
+  //Return if the matrix does not exist
+  if (!(this->have_matrix(mat_name)))
+    return;
+
+  _matrices[mat_name]->clear();
+  delete _matrices[mat_name];
+  _matrices[mat_name] = libmesh_nullptr;
```

Comment:
This leaves a NULL entry in the `_matrices` container that will cause `ImplicitSystem::have_matrix()` to return confusing values.